src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -199,46 +199,19 @@ public VpnService.Builder newBuilder() {
    * Helper method to build a TunnelConfig from a JSON object.
    *
    * @param tunnelId unique identifier for the tunnel.
-   * @param config JSON object containing TunnelConfig values.
+   * @param configString Config string containing TunnelConfig values.
    * @throws IllegalArgumentException if `tunnelId` or `config` are null.
    * @throws JSONException if parsing `config` fails.
    * @return populated TunnelConfig
    */
-  public static TunnelConfig makeTunnelConfig(final String tunnelId, final JSONObject config)
+  public static TunnelConfig makeTunnelConfig(final String tunnelId, final String configString)
```

Comment:
Revert to the JSONObject. You'll extract the proxy string from a field in the JSON.

Reply:
See _CordovaTunnel_ in cordova_main.ts
_start_ receives a  _ShadowsocksSessionConfig_, right? And now I pass the opaque string version rather than its original JSON format. 
Do we intend to introduce a new container config which wraps this proxy config [string] and pass that from Cordova layer?

Reply:
I believe we can get rid of the IS_REACHABLE call too. It's not really useful. We use it to determine the status of the Tunnel, but we can do that in a different way. It makes sense to have a separate PR to just remove IsReachable first.